src/python/pants/bsp/util_rules/targets.py

```diff
@@ -131,9 +137,49 @@ async def bsp_build_target_sources(request: SourcesParams) -> SourcesResult:
     return SourcesResult(items=tuple(si.sources_item for si in sources_items))
 
 
+# -----------------------------------------------------------------------------------------------
+# Dependency Sources Request
+# See https://build-server-protocol.github.io/docs/specification.html#dependency-sources-request
+# -----------------------------------------------------------------------------------------------
+
+
+class DependencySourcesHandlerMapping(BSPHandlerMapping):
+    method_name = "buildTarget/dependencySources"
+    request_type = DependencySourcesParams
+    response_type = DependencySourcesResult
+
+
+@rule
+async def bsp_dependency_sources(request: DependencySourcesParams) -> DependencySourcesResult:
+    return DependencySourcesResult(
```

Comment:
Is it worth adding these as empty? If so, it might be good to leave a TODO indicating that they're still subbed (or have them raise, maybe).

Reply:
Yes because then it lets me see how the IDE reacts to having these available. VSCode assumes that "Dependency Modules Request" is supported and ignores the capability bit for it.